notebooks/article_1.py

```diff
@@ -0,0 +1,1201 @@
+
+# In[10]:
+
+
+import matplotlib.pyplot as plt
+
+# Assuming 'data_df' is your DataFrame and it contains 'state_code' and 'status' columns
+grouped_data = data_df.groupby(['state_code', 'status']).size().unstack().fillna(0)
+
+# Sort the DataFrame by the sum of 'acquired' and 'closed' for each state and take the first 10
+grouped_data = grouped_data.sort_values(by=['acquired', 'closed'], ascending=False).head(10)
+
+# Convert the index to string to ensure proper handling as categorical data
+grouped_data.index = grouped_data.index.astype(str)
+
+# Create the stacked bar chart
+ax = grouped_data.plot(kind='bar', stacked=True, figsize=(15, 8), color=['skyblue', 'salmon'])
+plt.title('Number of Acquired vs. Closed Startups in Top 10 States',fontsize=20, pad=25, fontweight='bold')
+plt.xlabel('State Code', fontsize=18,labelpad=10)
+plt.ylabel('Number of Startups', fontsize=18,labelpad=20)
+plt.xticks(rotation=45)  # Rotates the state labels for better readability
+plt.legend(title='Status')
+
+# Define a threshold for displaying annotations
+threshold = 50  # Adjust the threshold as needed
+
+# Add text annotations
+for i, (state_code, row) in enumerate(grouped_data.iterrows()):
+    cumulative_height = 0
+    for status in ['acquired', 'closed']:  # Iterate in a specific order
+        count = row[status]
+        # Display annotation only if count is above the threshold
+        if count > threshold:
+            # Position text at the center of each segment
+            ax.text(i, cumulative_height + count / 2, f'{int(count):,}', 
+                    fontsize=8, color='black', ha='center', va='center')
+        cumulative_height += count
+
+# Slightly adjust the ylim to make room for text annotations
+ax.set_ylim(0, grouped_data.max().sum() * 1.1)  # Increase the y-axis limit
+plt.show()
+
+
+# The bar chart shows the following location trends:
+# 
+# - California (CA) stands out with the highest total number of startups, where the number of acquired startups is more than double the number closed.
+# 
+# - New York (NY) and Massachusetts (MA) also exhibit a pattern where more startups are being acquired than closed, suggesting a favorable startup environment in these states as well.
+#   
+# - In other states such as Washington (WA), Texas (TX), and Colorado (CO), the numbers of acquisitions and closures are closer, pointing to a more challenging environment.
+
+# We've seen how important visuals are for understanding data and identifying important discoveries.
+# 
+# After using the categorical `status` variable, we'll refine it by converting `acquired` and `closed` into numeric values to facilitate analysis. 
+# 
+# To accomplish this, we'll use the **mapping method**, which assigns a unique numerical value to each category.
+
+# In[11]:
+
+
+data_df['status'] = data_df.status.map({'acquired':1, 'closed':0})
+data_df.head(2)
+
+
+#  ### Missing Values
```

Comment:
The missing values should be place just after the "Handle irrelevant features" section.